PXC-3508 : SST with OpenSSL 1.1.1 and encrypt=2,3 fails with "E SSL_C…

kennt-percona opened the pull request:
…TX_set_tmp_dh" from socat

Issue
Using OpenSSL fixed a bug with the way it used/generated dhparams.  This means that
certificates generated using an older version of OpenSSL would now fail if
used with OpenSSL 1.1.1.

Solution
Explicitly set the dhparam option with socat to workaround the use of the
old certs.

kamil-holubicki reviewed (commented):
Looks good,
Do we have MTR for this?
Do we have Jenkins run for this?

kennt-percona commented:
https://pxc.cd.percona.com/job/pxc-5.7-param/96/#showFailuresLink
The tests are still failing for the newer distros (focal, centos8, buster), but they don't use SSL, so the tests are failing for a different reason.

kennt-percona committed:
PXC-3508 : SST with OpenSSL 1.1.1 and encrypt=2,3 fails with "E SSL_CTX_set_tmp_dh" from socat
cherry-picked from commit 6722cd1a76b3f17b4863dbd7e1b3f499b4e4b029

Issue
Using OpenSSL fixed a bug with the way it used/generated dhparams.  This means that
certificates generated using an older version of OpenSSL would now fail if
used with OpenSSL 1.1.1.

Solution
Explicitly set the dhparam option with socat to workaround the use of the
old certs.

kennt-percona commented:
https://pxc.cd.percona.com/job/pxc-8.0-param/182/#showFailuresLink

The tests that were failing on newer distros are no longer failing.
galera_sst_xtrabackup-v2_encrypt_with_key
galera_sst_xtrabackyp-v2_keyring
